app/javascript/chat/ChatChannelSettings/ChatChannelSettingsSection.jsx

```diff
@@ -0,0 +1,106 @@
+import { h } from 'preact';
+import PropTypes from 'prop-types';
+
+import ModSection from './ModSection';
+import PersonalSettings from './PersonalSetting';
+import LeaveMembershipSection from './LeaveMembershipSection';
+import ModFaqSection from './ModFaqSection';
+import ChannelDescriptionSection from './ChannelDescriptionSection';
+import ChatChannelMembershipSection from './ChatChannelMembershipSection';
+
+const ChatChannelSettingsSection = ({
+  channelDiscoverable,
+  updateCurrentMembershipNotificationSettings,
+  handleleaveChatChannelMembership,
+  handlePersonChatChennelSetting,
+  handleChannelDescriptionChanges,
+  handleChannelDiscoverableStatus,
+  handleDescriptionChange,
+  handleChatChannelInvitations,
+  handleInvitationUsernames,
+  toggleScreens,
+  removeMembership,
+  chatChannelAcceptMembership,
+  channelDescription,
+  chatChannel,
+  currentMembership,
+  activeMemberships,
+  pendingMemberships,
+  requestedMemberships,
+  invitationUsernames,
+  showGlobalBadgeNotification,
```

Comment:
Just an observation, but there is quite a lot of props for this component. Membership and Channel props are being split up in the parent component. It would make sense to keep them grouped like they were in the original entities returned in the JSON payload.

Also, prefixing or using the word chat in props in here is not necessary. The name of the component they are being passed into `<ChatChannelSettingsSection ...   />` gives enough context that using `channel` is good enough for readability. e.g. `handleChatChannelInvitations` -> `handleChannelInvitations`, `handleInvitations` or even `processInvitations`

```suggestion
  updateCurrentMembershipNotificationSettings,
  handleleaveChatChannelMembership,
  handlePersonChatChennelSetting,
  handleChannelDescriptionChanges,
  handleChannelDiscoverableStatus,
  handleDescriptionChange,
  handleChatChannelInvitations,
  handleInvitationUsernames,
  toggleScreens,
  removeMembership,
  channel,
  memberships,
  invitationUsernames,
  showGlobalBadgeNotification,
```